src/condor_credd/condor_credmon_oauth/condor_vault_storer

```diff
@@ -150,10 +167,21 @@ but make sure no other job is using them."
     if [ -n "$BTOKEN" ]; then
         OPTS=("-o" "$BTOKEN-$SERVICE" "${OPTS[@]}")
     fi
-    OPTS=("--vaulttokenttl=28d" "${OPTS[@]}" "--vaulttokeninfile=$VTOKEN-$SERVICE" "--vaulttokenfile=/dev/stdout" "--showbearerurl")
+
+    if $IS_POMS_ANALYSIS; then
+        OPTS=("--vaulttokenttl=28d" "${OPTS[@]}" "--vaulttokeninfile=/dev/stdin" "--vaulttokenfile=/dev/stdout" "--nokerberos"  "--showbearerurl" "$CREDKEY")
+    else
+        OPTS=("--vaulttokenttl=28d" "${OPTS[@]}" "--vaulttokeninfile=$VTOKEN-$SERVICE" "--vaulttokenfile=/dev/stdout" "--showbearerurl")
+    fi
+
     verbose "Attempting to get tokens for $SERVICE"
     # First attempt to get tokens quietly without oidc
-    CRED="`htgettoken "${OPTS[@]}" --nooidc ${VERBOSE:--q}`"
+    if $IS_POMS_ANALYSIS; then
+        CRED="`cat $VTOKEN | htgettoken "${OPTS[@]}"  ${VERBOSE:--q}`"
```

Comment:
Do we want to also have `--nooidc` here?

Reply:
Yes, and why is `--vaulttokeninfile=/dev/stdin` needed if you're just going to cat a file into it htgettoken?  Why not just have htgettoken read directly from the file?